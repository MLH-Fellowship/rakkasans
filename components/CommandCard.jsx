import React from 'react';
import {
  TouchableOpacity, Text, StyleSheet, Image, View,
} from 'react-native';

import Colors from '../constants/Colors';
import Dimensions from '../constants/Dimensions';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: Colors.white,
    marginTop: 35,
    marginHorizontal: 20,
    alignItems: 'center',
    borderRadius: 5,
    borderWidth: 1,
    borderColor: Colors.primary,
  },
  image: {
    borderTopLeftRadius: 5,
    borderBottomLeftRadius: 5,
    width: 75,
    height: 100,
    marginRight: 20,
  },
  titleContainer: {
    flex: 1,
  },
  text: {
    color: Colors.darkGray,
    fontSize: 20,
  },
});

export default function CommandCard({ source, title, onPress }) {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image style={styles.image} resizeMode="cover" source={source} />
      <View style={styles.titleContainer}>
        <Text style={[styles.text, Dimensions.font]}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
}
