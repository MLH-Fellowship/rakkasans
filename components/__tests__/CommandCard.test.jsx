import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Image, Text } from 'react-native';
import renderer from 'react-test-renderer';

import CommandCard from '../CommandCard';

import Copic from '../../assets/images/COPIC.jpeg';

jest.useFakeTimers();
configure({ adapter: new Adapter() });

describe('CommandCard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CommandCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with an image', () => {
    const wrapper = shallow(
      <CommandCard>
        <Image src={Copic} />
        <Text>COL Brandon Teague</Text>
      </CommandCard>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles onPress', () => {
    const onPress = jest.fn();

    const wrapper = shallow(
      <CommandCard onPress={onPress}>
        <Image src={Copic} />
        <Text>COL Brandon Teague</Text>
      </CommandCard>,
    );

    wrapper
      .find('TouchableOpacity')
      .first()
      .props()
      .onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
