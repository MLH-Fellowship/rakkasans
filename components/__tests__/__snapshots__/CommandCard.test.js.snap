// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#020a41",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "marginTop": 35,
      "opacity": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    style={
      Object {
        "borderBottomLeftRadius": 5,
        "borderTopLeftRadius": 5,
        "height": 100,
        "marginRight": 20,
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "fontSize": 20,
          },
          Object {
            "fontFamily": "andale-mono",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`CommandCard renders correctly with an image 1`] = `ShallowWrapper {}`;
